### PR TITLE
Add \ for Sym/Tri/Bi/Diagonal and support non-commutative numbers

### PR DESCRIPTION
**Pull request by dlfivefifty**

### CODE REVIEW

File: stdlib/LinearAlgebra/src/tridiag.jl [hunk 1]
```diff
@@ -208,6 +208,7 @@ end
 *(A::SymTridiagonal, B::Number) = SymTridiagonal(A.dv*B, A.ev*B)
 *(B::Number, A::SymTridiagonal) = A*B
 /(A::SymTridiagonal, B::Number) = SymTridiagonal(A.dv/B, A.ev/B)
+\(B::Number, A::Tridiagonal) = A/B
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/tridiag.jl:211]:
```suggestion
\(B::Number, A::SymTridiagonal) = A/B
```

**Comment by dkarrasch** [comment]
Are these rules correct also in the non-commutative case? I see we're using this "flip the order" pattern frequently...

**Comment by dlfivefifty** [comment]
I agree it doesn't look good for non-commutativit, but this is true also for `*`. Let me push a better variant and see if tests pass.

**Comment by dlfivefifty** [comment]
Do you know if there's a non-commutative type `<: Number` in the tests I can use? Otherwise I can add a quick-and-dirty `Quaternion`.

**Comment by dkarrasch** [comment]
We have a `Quaternion` type in the testhelpers folder.

**Comment by dkarrasch** [comment]
I think `Diagonal` is missing the same `\` method, BTW.

**Comment by dlfivefifty** [comment]
OK I added the overload for Diagonal too

**Comment by dlfivefifty** [comment]
I believe the test failure is not related.

**Review by dkarrasch — APPROVED** [review]

**Comment by dkarrasch** [comment]
Let's run CI again and then merge.

**Comment by dkarrasch** [comment]
Only failure is a timeout, but way after the LinearAlgebra tests.

**Pull request merged**